canal/sync.go

```diff
@@ -119,42 +111,25 @@ func (c *Canal) runSyncBinlog() error {
 				return errors.Trace(err)
 			}
 		case *replication.QueryEvent:
-			var (
-				mb    [][]byte
-				db    []byte
-				table []byte
-			)
-			regexps := []regexp.Regexp{*expCreateTable, *expAlterTable, *expRenameTable, *expDropTable, *expTruncateTable}
-			for _, reg := range regexps {
-				mb = reg.FindSubmatch(e.Query)
-				if len(mb) != 0 {
-					break
-				}
+			if e.GSet != nil {
+				c.master.UpdateGTIDSet(e.GSet)
 			}
-			mbLen := len(mb)
-			if mbLen == 0 {
+			stmts, _, err := c.parser.Parse(string(e.Query), "", "")
+			if err != nil {
+				log.Errorf("parse query(%s) err %v", e.Query, err)
```

Comment:
is `continue` ok here @siddontang .
I always think old logic `if mbLen == 0  then continue` is mainly used to filter `Begin` and `Dummy` events, not for DDL that parser don't support, right?

Reply:
fine, if parse error, may return error is better.